test/common/upstream/maglev_lb_test.cc

```diff
@@ -224,18 +224,18 @@ TEST_F(MaglevLoadBalancerTest, LocalityWeightedGlobalPanic) {
   // maglev: i=5 host=127.0.0.1:91
   // maglev: i=6 host=127.0.0.1:91
   // maglev: i=7 host=127.0.0.1:90
-  // maglev: i=8 host=127.0.0.1:91
+  // maglev: i=8 host=127.0.0.1:90
   // maglev: i=9 host=127.0.0.1:91
-  // maglev: i=10 host=127.0.0.1:91
+  // maglev: i=10 host=127.0.0.1:90
   // maglev: i=11 host=127.0.0.1:91
-  // maglev: i=12 host=127.0.0.1:91
+  // maglev: i=12 host=127.0.0.1:90
   // maglev: i=13 host=127.0.0.1:90
   // maglev: i=14 host=127.0.0.1:91
   // maglev: i=15 host=127.0.0.1:90
   // maglev: i=16 host=127.0.0.1:91
   LoadBalancerPtr lb = lb_->factory()->create();
-  const std::vector<uint32_t> expected_assignments{1, 0, 0, 1, 0, 1, 1, 0, 1,
-                                                   1, 1, 1, 1, 0, 1, 0, 1};
+  const std::vector<uint32_t> expected_assignments{1, 0, 0, 1, 0, 1, 1, 0, 0,
+                                                   1, 0, 1, 0, 0, 1, 0, 1};
```

Comment:
Thanks for this fix; I'll do a more significant review in a bit, but to get started, I think it would be good to add the extreme locality weighting test that we had for ring hash LB for Maglev as well, to ensure we see similar correct behavior.

Reply:
Done. The expected behavior is a bit different, but it's still a good test case.